x-pack/solutions/security/plugins/security_solution/public/flyout/attack_details/hooks/use_find_attack_details_by_id/index.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  API_VERSIONS,
+  ATTACK_DISCOVERY_FIND,
+  ATTACK_DISCOVERY_INTERNAL_FIND,
+  transformAttackDiscoveryAlertFromApi,
+} from '@kbn/elastic-assistant-common';
+import type { QueryObserverResult, RefetchOptions, RefetchQueryFilters } from '@kbn/react-query';
+import { useQuery } from '@kbn/react-query';
+import { useCallback, useRef } from 'react';
+
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import * as i18n from './translations';
+import { useKibanaFeatureFlags } from '../../../../attack_discovery/pages/use_kibana_feature_flags';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+interface Props {
+  id: string;
+  http: HttpSetup;
+  isAssistantEnabled: boolean;
+  refetchOnWindowFocus?: boolean;
+}
+
+type AttackDiscoveryInternalItem =
+  | AttackDiscoveryFindInternalResponse
+  | (AttackDiscoveryFindResponse extends { data: (infer U)[] }
+      ? ReturnType<typeof transformAttackDiscoveryAlertFromApi>
+      : unknown);
+
+export type AttackDiscoveryItem = AttackDiscoveryFindResponse extends { data: (infer U)[] }
+  ? ReturnType<typeof transformAttackDiscoveryAlertFromApi>
+  : unknown;
+
+interface UseFindAttackDiscoveryById {
+  cancelRequest: () => void;
+  data: AttackDiscoveryItem | undefined;
+  error: unknown | undefined;
+  isLoading: boolean;
+  refetch: <TPageData>(
+    options?: (RefetchOptions & RefetchQueryFilters<TPageData>) | undefined
+  ) => Promise<QueryObserverResult<AttackDiscoveryInternalItem | undefined, unknown>>;
+  status: 'error' | 'idle' | 'loading' | 'success';
+}
+
+/**
+ * Fetch a single attack discovery by document id using the FIND endpoint.
+ * Uses the public/internal FIND route depending on the feature flag and
+ * maps the public API shape into the internal UI shape.
+ */
+export const useFindAttackDetailsById = ({
+  id,
+  http,
+  isAssistantEnabled,
+  refetchOnWindowFocus = false,
+}: Props): UseFindAttackDiscoveryById => {
+  const { addError } = useAppToasts();
+  const { attackDiscoveryPublicApiEnabled } = useKibanaFeatureFlags();
+  const abortController = useRef(new AbortController());
+
+  const cancelRequest = useCallback(() => {
+    abortController.current.abort();
+    abortController.current = new AbortController(); // LOCAL MUTATION
+  }, []);
+
+  const route = attackDiscoveryPublicApiEnabled
+    ? ATTACK_DISCOVERY_FIND
+    : ATTACK_DISCOVERY_INTERNAL_FIND;
+
+  const version = attackDiscoveryPublicApiEnabled
+    ? API_VERSIONS.public.v1
+    : API_VERSIONS.internal.v1;
+
+  const queryFn = useCallback(async () => {
+    const baseQuery = {
+      ids: [id],
+      page: 1,
+      per_page: 1,
+    };
+
+    if (attackDiscoveryPublicApiEnabled) {
+      return http.fetch<AttackDiscoveryFindResponse>(route, {
+        method: 'GET',
+        version,
+        query: {
+          ...baseQuery,
+          enable_field_rendering: true, // always true to enable rendering fields using the `{{ user.name james }}` syntax
```

Comment:
Do we need `enable_field_rendering` and `with_replacements` for our usage in the flyout? I'm not sure what those actually do...